src/app/public/modules/phone-field/phone-field.component.spec.ts

```diff
@@ -0,0 +1,903 @@
+
+    describe('initialization', () => {
+
+      it('should initialize the default country correctly', fakeAsync(() => {
+        component.defaultCountry = 'us';
+        fixture.detectChanges();
+        tick();
+        fixture.detectChanges();
+
+        expect(nativeElement.querySelector('input').placeholder)
+          .toBe(component.phoneFieldComponent.countryData
+            .find(country => country.iso2 === 'us').exampleNumber);
+      }));
+
+      it('should initialize without a default country', fakeAsync(() => {
+        fixture.detectChanges();
+        tick();
+        fixture.detectChanges();
+
+        expect(nativeElement.querySelector('input').placeholder)
+          .toBe(component.phoneFieldComponent.countryData[0].exampleNumber);
+      }));
+
+      it('should handle initializing with number', fakeAsync(() => {
+        component.modelValue = '8675555309';
+        fixture.detectChanges();
+        tick();
+        fixture.detectChanges();
+
+        expect(nativeElement.querySelector('input').value).toBe('8675555309');
+      }));
+
+      it('should handle undefined initialization', fakeAsync(() => {
+        fixture.detectChanges();
+        tick();
+        fixture.detectChanges();
+
+        expect(nativeElement.querySelector('input').value).toBe('');
+
+        expect(nativeElement.querySelector('input')).not.toHaveCssClass('ng-invalid');
+      }));
+
+      it('should throw an error if directive is added in isolation', function () {
+        try {
+          component.showInvalidDirective = true;
+          fixture.detectChanges();
+        } catch (err) {
+          expect(err.message).toContain('skyPhoneFieldInput');
+        }
+      });
+
+    });
+
+    describe('input change', () => {
+
+      it('should handle input change with a string with the expected format', fakeAsync(() => {
+        fixture.detectChanges();
+        tick();
+        fixture.detectChanges();
+
+        setInput(nativeElement, '8675555309', fixture);
+        expect(nativeElement.querySelector('input').value).toBe('8675555309');
+        expect(component.modelValue).toEqual('(867) 555-5309');
+      }));
+
+      it('should handle input change with a string with the an unexpected format', fakeAsync(() => {
+        fixture.detectChanges();
+        tick();
+        fixture.detectChanges();
+
+        setInput(nativeElement, '86755-5-309', fixture);
+        expect(nativeElement.querySelector('input').value).toBe('86755-5-309');
+        expect(component.modelValue).toEqual('86755-5-309');
+      }));
+
+    });
+
+    describe('model change', () => {
+
+      it('should handle model change with a valid number', fakeAsync(() => {
+        fixture.detectChanges();
+        component.modelValue = '8675555309';
+        fixture.detectChanges();
+        tick();
+        fixture.detectChanges();
+
+        expect(nativeElement.querySelector('input').value).toBe('8675555309');
+      }));
+
+    });
+
+    describe('validation', () => {
```

Comment:
Maybe add a unit test to make sure that the phone field `FormControlAccessor` is setting the appropriate custom `skyPhoneField` error when an invalid phone number is provided?

Reply:
I already had a lot of tests that were checking for a valid state. I updated all of them to also look at the errors being returned (or not returned) appropriately.